File: backend/app/grade_config.py
# kelas -> {jumlah_soal, durasi_menit}
GRADE_CONFIG: dict[int, dict[str, int]] = {
    1: {"jumlah_soal": 20, "durasi_menit": 60},
    2: {"jumlah_soal": 20, "durasi_menit": 60},
    3: {"jumlah_soal": 25, "durasi_menit": 60},
    4: {"jumlah_soal": 25, "durasi_menit": 60},
    5: {"jumlah_soal": 30, "durasi_menit": 60},
    6: {"jumlah_soal": 30, "durasi_menit": 60},
}

DEFAULT_GRADE_CONFIG = {"jumlah_soal": 30, "durasi_menit": 90}


def get_config(grade: int) -> dict[str, int]:
    return GRADE_CONFIG.get(grade, DEFAULT_GRADE_CONFIG)

File: backend/internal/gradeconfig/gradeconfig.go
// Package gradeconfig holds the per-grade defaults used when an exam type
// doesn't specify its own question count / duration.
package gradeconfig

type Config struct {
	JumlahSoal  int
	DurasiMenit int
}

var byGrade = map[int]Config{
	1: {JumlahSoal: 20, DurasiMenit: 60},
	2: {JumlahSoal: 20, DurasiMenit: 60},
	3: {JumlahSoal: 25, DurasiMenit: 60},
	4: {JumlahSoal: 25, DurasiMenit: 60},
	5: {JumlahSoal: 30, DurasiMenit: 60},
	6: {JumlahSoal: 30, DurasiMenit: 60},
}

var defaultConfig = Config{JumlahSoal: 30, DurasiMenit: 90}

func Get(grade int) Config {
	if c, ok := byGrade[grade]; ok {
		return c
	}
	return defaultConfig
}
